src/SmartComponents/Inventory/applications/compliance/SystemRulesTable.js

```diff
@@ -1,181 +1,364 @@
+
+        return window.insights.chrome.auth.getUser()
+        .then(() => {
+            return remediationsApi.getResolutionsBatch(ruleIds)
+            .then((response) => {
+                if (!response.ok) {
+                    // If remediations doesn't respond, inject no fix available
+                    return {};
+                }
+
+                return response.json();
+            });
+        }).then(response => newRows.map(({ cells, ...row }) => ({
+            ...row,
+            cells: [
+                ...cells,
+                ...this.isParent(row, cells) ? [ this.remediationAvailable(response[`compliance:${refIds[cells[0]]}`]) ] : []
+            ]
+        })));
+    }
+
+    selectAll = (rows, selected) => {
+        return rows.map(row => (
+            {
+                ...row,
+                ...this.isParent(row) ? { selected } : {}
+            }
+        ));
+    }
+
+    onSelect = (_event, selected, key) => {
+        const { page, itemsPerPage } = this.state;
+        let { currentRows, rows } = this.state;
+        if (key === -1) {
+            rows = this.selectAll(rows, selected);
+            currentRows = this.selectAll(currentRows, selected);
         } else {
-            openNodes = [
-                ...openNodes,
-                Number(rowKey) + 1
-            ];
+            // One rule was selected
+            const index = ((page - 1) * itemsPerPage * 2) + Number(key);
+            rows[index].selected = selected;
+            currentRows[key].selected = selected;
         }
 
-        this.setState({
-            openNodes,
-            rows
+        this.setState({ rows, currentRows });
+    }
+
+    calculateParent = ({ profile }, { title, severity, compliant }) => ({
+        isOpen: false,
+        cells: [
+            title,
+            profile,
+            severity,
+            (compliant ? <CheckCircleIcon className='ins-u-passed' /> :
+                <ExclamationCircleIcon className='ins-u-failed' />)
+        ]
+    })
+
+    calculateChild = ({ description, rationale }, key) => ({
+        parent: key * 2,
+        cells: [{
+            title: (
+                <React.Fragment key={ key }>
+                    <Stack id={ `rule-description-${key}` } gutter="md">
+                        <StackItem><b>Description</b></StackItem>
+                        <StackItem isMain>{ description }</StackItem>
+                    </Stack>
+                    <br/>
+                    <Stack id={ `rule-rationale-${key}` } gutter="md">
+                        <StackItem><b>Rationale</b></StackItem>
+                        <StackItem isMain>{ rationale }</StackItem>
+                    </Stack>
+                </React.Fragment>
+            ) }]
+    })
+
+    rulesToRows = (profileRules) => {
+        const refIds = {};
+        const rows = flatMap(profileRules, (profileRule) => profileRule.rules.flatMap((rule, key) => {
```

Comment:
I don't think we can use `Array.protoype.flatMap` because it's not well supported. Can you use `flatMap` from lodash please?
```JS
const rows = flatMap(profileRules, (profileRule) => flatMap(profileRule.rules, (rule, key) => {
```